dartagnan/src/test/java/com/dat3m/dartagnan/c/VMMLFDSTest.java

```diff
@@ -0,0 +1,78 @@
+package com.dat3m.dartagnan.c;
+
+import com.dat3m.dartagnan.configuration.Arch;
+import com.dat3m.dartagnan.configuration.Property;
+import com.dat3m.dartagnan.utils.Result;
+import com.dat3m.dartagnan.utils.rules.Provider;
+import com.dat3m.dartagnan.utils.rules.Providers;
+import com.dat3m.dartagnan.verification.solving.RefinementSolver;
+import com.dat3m.dartagnan.wmm.Wmm;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.EnumSet;
+
+import static com.dat3m.dartagnan.configuration.Arch.IMM;
+import static com.dat3m.dartagnan.configuration.Property.*;
+import static com.dat3m.dartagnan.utils.ResourceHelper.getTestResourcePath;
+import static com.dat3m.dartagnan.utils.Result.*;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
+public class VMMLFDSTest extends AbstractCTest {
+
+    public VMMLFDSTest(String name, Arch target, Result expected) {
+        super(name, target, expected);
+    }
+
+    @Override
+    protected Provider<String> getProgramPathProvider() {
+        return Provider.fromSupplier(() -> getTestResourcePath("lfds/" + name + ".ll"));
+    }
+
+    @Override
+    protected long getTimeout() {
+        return 600000;
+    }
+
+    @Override
+    protected Provider<EnumSet<Property>> getPropertyProvider() {
+        return Provider.fromSupplier(() -> EnumSet.of(PROGRAM_SPEC, LIVENESS, CAT_SPEC));
+    }
+
+    @Override
+    protected Provider<Integer> getBoundProvider() {
+        return Provider.fromSupplier(() -> 2);
+    }
```

Comment:
Just a little thing nitpick: You actually don't need the `Provider.fromSupplier` wrapper but you can write 
`return () -> 2` directly.